### PR TITLE
pre-signed URL upload for better security

### PR DESCRIPTION
**Pull request by amjedidiah**

### CODE REVIEW

**Review by BSalaeddin — COMMENTED** [review]
### Variable Naming Improvements suggestions

1. **`files`**: Rename to `uploadedFiles` or `selectedFiles` in components and hooks that expose it to developers. This will make its purpose clearer when returned by hooks or used as props in components (e.g., in `UpupUploader.tsx`).

2. **`onChange`**: Rename to `onFileSelectionChange` for props exposed by components to better describe its role in handling file selection updates.

3. **`toBeCompressed`**: Rename to `shouldCompress` for component props to better reflect its boolean nature and improve clarity for developers configuring the component.

4. **`keys`**: Rename to `uploadedFileKeys` if returned by hooks or methods, to clarify that it represents the identifiers of successfully uploaded files.

**Review by MedAmine1212 — APPROVED** [review]
lgtm

File: src/components/UpupUploader/DropZone.tsx [hunk 1]
```diff
@@ -66,7 +66,7 @@ const DropZone: FC<Props> = ({
                         return multiple
                             ? [...prev, ...acceptedFiles]
                             : // only one file
-                              [acceptedFiles[0]]
+                              [acceptedFiles[0]].filter(Boolean)
                     })
```
Comment (MedAmine1212) [inline src/components/UpupUploader/DropZone.tsx]: Could you explain this please @amjedidiah why did you add the .filter(Boolean) ?

Comment (amjedidiah) [reply src/components/UpupUploader/DropZone.tsx]: @MedAmine1212, updated to be more readable and safe.

**Review by BSalaeddin — APPROVED** [review]
